### PR TITLE
Fix unguarded `ResizeObserver` access

### LINKED ISSUE #794: Unguarded `ResizeObserver` access
## Description

https://sentry.io/share/issue/3a2ae88e6acc49e5bf237070c06a3a17/

Mobile Safari 15.3 is a current version of that browser and it _should_ have ResizeObserver: https://caniuse.com/?search=ResizeObserver

Not sure what's causing it to not be defined, might be an anti-fingerprinting technique or a permissions issue. Either way, we should guard against this as it's causing errors on a regular basis and should be trivial to fix. Mobile browsers rarely resize anyway.

## Resolution

- [x] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #794 by @sarayourfriend 

## Description

Prevents ResizeObserver from being accessed if it does not exist. I don't think we need to polyfill it or anything in these cases and we can pretty gracefully degrade the experience when it's not around without hurting anything.

## Testing Instructions

Do not try to test this on Firefox by disabling resize observer. It will completely destroy your Firefox personalized settings.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by zackkrida — APPROVED** [review]
> Do not try to test this on Firefox by disabling resize observer. It will completely destroy your Firefox personalized settings.

Ugh, I imagine you learned this the hard way. Looks good. The homepage logo error fix is welcome too.

**Comment by sarayourfriend** [comment]
I did indeed @zackkrida

https://bugzilla.mozilla.org/show_bug.cgi?id=1753408

File: src/pages/index.vue [hunk 1]
```diff
@@ -6,7 +6,11 @@
     <header
       class="flex-grow w-full lg:w-auto lg:min-w-[32rem] xl:min-w-[64rem] box-border flex flex-col justify-between lg:justify-center"
     >
-      <VLogoButton class="lg:hidden ms-3" :auto-resize-logo="false" />
+      <VLogoButton
```
Comment (obulat) [inline src/pages/index.vue:9]: With the change in #816 this change is no longer necessary.

**Review by obulat — APPROVED** [review]
It's amazing how such seemingly small changes can have so much history and wreak so much havoc :)

File: src/pages/index.vue [hunk 2]
```diff
@@ -6,7 +6,11 @@
...
+        class="lg:hidden ms-3"
+        :auto-resize-logo="false"
+        :is-search-route="false"
+      />
```
Comment (sarayourfriend) [inline src/pages/index.vue:13]:
```suggestion
      <VLogoButton class="lg:hidden ms-3" :auto-resize-logo="false" />
```

**Pull request merged**